### PR TITLE
[GAIA Compiler] Build algebra layer structures for expressions

### PR DESCRIPTION
**Pull request by shirly121**
## What do these changes do?
1. implement `variable(..)` in `GraphBuilder.java` to build algebra layer structures for expressions.
2. add `NameOrId` to denote property key as name or id in `RexGraphVariable`.

## Related issue number

Fixes

### CODE REVIEW

File: interactive_engine/compiler/src/test/java/com/alibaba/graphscope/common/ir/ExpressionTest.java [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    }
+
+    @Test
+    public void variable_3_test() {
+        try {
+            builder.source(mockSourceConfig(null)).variable("a");
+        } catch (IllegalArgumentException e) {
+            return;
+        }
+        Assert.fail("tag 'a' should not exist");
```
Comment (longbinlai) [inline interactive_engine/compiler/src/test/java/com/alibaba/graphscope/common/ir/ExpressionTest.java:70]: `tag 'a' does not exist` is a better description.

File: interactive_engine/compiler/src/test/java/com/alibaba/graphscope/common/ir/ExpressionTest.java [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+        Assert.assertEquals(equal.getType().getSqlTypeName(), SqlTypeName.BOOLEAN);
+        Assert.assertEquals("=(a.age, 'X')", equal.toString());
+    }
+
+    // a.age + 10 > 30
+    @Test
+    public void equal_plus_test() {
+        RexNode var = builder.source(mockSourceConfig("a")).variable("a", "age");
+        RexNode plus = builder.call(GraphStdOperatorTable.PLUS, var, builder.literal(10));
+        RexNode equal = builder.call(GraphStdOperatorTable.EQUALS, plus, builder.literal(30));
```
Comment (longbinlai) [inline interactive_engine/compiler/src/test/java/com/alibaba/graphscope/common/ir/ExpressionTest.java:121]: According to your comment, this should be `GreaterThan` ?

File: interactive_engine/compiler/src/test/resources/schema/modern.json [hunk 3]
```diff
@@ -76,7 +76,7 @@
           "comment": "name"
         },
         {
-          "id": 3,
+          "id": 4,
```
Comment (longbinlai) [inline interactive_engine/compiler/src/test/resources/schema/modern.json:79]: Why these ids changed?

File: interactive_engine/compiler/src/test/java/com/alibaba/graphscope/common/ir/ExpressionTest.java (hunk 1 above)
Comment (shirly121) [reply interactive_engine/compiler/src/test/java/com/alibaba/graphscope/common/ir/ExpressionTest.java]: done

File: interactive_engine/compiler/src/test/java/com/alibaba/graphscope/common/ir/ExpressionTest.java (hunk 2 above)
Comment (shirly121) [reply interactive_engine/compiler/src/test/java/com/alibaba/graphscope/common/ir/ExpressionTest.java:121]: fix comments

**Review by longbinlai — APPROVED** [review]
LGTM

**Pull request merged**